### DISCUSSION TITLE
Last Valid Value

### DISCUSSION THREAD

**Discussion by firesgc** [jomjol/AI-on-the-edge-device#1150] [Q&A] [answered]
Hello!

is it possible to return the last valid value in the REST Api when the detection fails? At the moment it's just an empty string.
If you don't want to break compatibility, maybe you could add a new field called "LastValidValue"?

Thanks a lot!

current return value:
{
"main":
  {
    "value": "",
    "raw": "20148.N0",
    "error": "Neg. Rate - Read:  - Raw: 20148.N0 - Pre: 20148.46 ",
    "rate": "",
    "timestamp": "2022-10-08T14:31:32"
  }
}

**Comment by caco3** [comment]
isn't "Pre" what you are looking for?

**Reply by firesgc** [reply]
Yes, but it's in an error string, and I'm not sure it's guaranteed to always contain "Pre:" (an error string can end up being anything).

The LastValidValue is always the last valid value.

If there is no error, LastValidValue would be identical to value. 
But in my JSON parser in HomeAssistant, I can be sure that LastValidValue always returns a correct value. Otherwise, I need to add some logic like "if value is empty, try parsing the error string and hope it contains Pre:".

**Reply by caco3** [reply]
ah, right, we should extend that.

**Reply by caco3** [reply]
https://github.com/jomjol/AI-on-the-edge-device/issues/1154

**Comment by caco3** [comment, marked as answer]
now contained in rolling
